### ISSUE TITLE
Outputs from previous actions are lost

### ISSUE THREAD

**Issue by carolynvs** [deislabs/porter#836]
## Describe the bug
Old outputs are overwritten by new actions.

## To Reproduce
1. Run an action that generates a bundle level output
2. Run another action that does not generate that same named output
3. Note that the output from the first action is no longer persisted in the claim

## Expected behavior
The original output should still be persisted and available.

## Version
Copy the output of `porter version` below

`porter v0.22.0-beta.1 (8f3e429)`

**Comment by carolynvs** [comment]
There are two scenarios for this issue

1. install generated output foo and upgrade doesn’t generate foo at all. I don’t want to lose foo completely in this case.
2. install generated foo with certain contents. upgrade also generates output foo with new contents. I want to see the 1 output by the end named foo with the updated contents

**Comment by vdice** [comment]
> 1. install generated output foo and upgrade doesn’t generate foo at all. I don’t want to lose foo completely in this case.

This may require amendments to the [claim spec](https://github.com/cnabio/cnab-spec/blob/master/400-claims.md) -- which of course is reasonable.  As it stands, I think the claim's outputs section may not yet have been designed to persist all outputs across all actions.  Rather, I think it currently is designed to capture only the outputs declared by the given action that has just run.  Therefore, if install generates `foo` but upgrade does not, I think no longer seeing output `foo` in the claim after upgrade runs is as currently designed.

The spec currently says `outputs: Key/value pairs that were created by the operation. These are stored so that the user can access them after the operation completes`.  Again, it doesn't necessarily mention any intention of persisting outputs across operations (actions), and perhaps this is where we could come in and suggest changes if we deem it appropriate.

Alternatively, regardless of the state of the claim spec, we may wish to introduce Porter-specific functionality to persist outputs across actions, perhaps via a local directory or some other mechanism.  This would be reasonable as well.

**Comment by vdice** [comment]
> 2. install generated foo with certain contents. upgrade also generates output foo with new contents. I want to see the 1 output by the end named foo with the updated contents

I may be misunderstanding this one, as I seem to be seeing an output generated by both actions being updated properly:

```
$ p install --param a=AAA
...

Outputs:

a = AAA
b = b
c = c
execution completed successfully!

 $ p instance output show a
AAA

 $ p upgrade --param a=BBB
...

Outputs:

a = BBB
b = b
c = c
execution completed successfully!

 $ p instance output show a
BBB
```
(Using https://github.com/deislabs/porter-terraform/tree/master/examples/basic-tf-example above)

**Comment by carolynvs** [comment]
Here is the gist that I am using to reproduce. After running upgrade, the output from install is gone and doesn't show up anymore using `porter instance outputs list`

https://gist.github.com/carolynvs/73bff79864ebeef0c6c0ca586553186a

**Comment by vdice** [comment]
Ah, thank you for the gist.  Indeed, I _think_ the behavior currently is due to https://github.com/deislabs/porter/issues/836#issuecomment-566699416.  (Meaning, not yet supported in the Claims spec and not yet exposed as extra-spec/Porter-specific functionality.)  WDYT?

**Comment by carolynvs** [comment]
Closed by #1065
